### PR TITLE
Add group IAM

### PR DESCRIPTION
**Pull request by sourced-vince**

### CODE REVIEW

File: community/cloud-foundation/templates/project/project.py [hunk 1]
```diff
@@ -11,12 +11,15 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
```
Comment (sourced-pavel) [inline community/cloud-foundation/templates/project/project.py]: Doesn't pass pylint check:

```
C: 98, 0: Wrong hanging indentation before block (add 4 spaces).
        properties.get('usageExportBucket') and
        ^   | (bad-continuation)
C: 99, 0: Wrong hanging indentation before block (add 4 spaces).
        'storage-component.googleapis.com' not in apis
        ^   | (bad-continuation)
C:175, 0: Wrong hanging indentation before block (add 4 spaces).
    context,
    ^   | (bad-continuation)
C:176, 0: Wrong hanging indentation before block (add 4 spaces).
    dependencies,
    ^   | (bad-continuation)
C:177, 0: Wrong hanging indentation before block (add 4 spaces).
    members_list
    ^   | (bad-continuation)
C:174, 0: Function name "create_shared_vpc_subnet_iam_for_service_accounts" doesn't conform to snake_case naming style (invalid-name)
C:213, 0: Function name "create_service_accounts_and_set_project_iam" doesn't conform to snake_case naming style (invalid-name)

```

Comment (sourced-vince) [reply community/cloud-foundation/templates/project/project.py]: This is an example of where YAPF and PYLINT aren't on the same page. I can modify it but it'll complain again the next time some runs YAPF and PYLINT.

File: community/cloud-foundation/templates/project/project.py [hunk 2]
```diff
@@ -251,18 +250,19 @@ def create_service_accounts(context):
             }
         )
 
+    # Build the group bindings for project IAM permissions
+    for group in context.properties['groups']:
+        group_name = "group:{}".format(group['name'])
```
Comment (sourced-pavel) [inline community/cloud-foundation/templates/project/project.py]: Inconsistent quotes - `"`. Elsewhere it's `'`

Comment (sourced-vince) [reply community/cloud-foundation/templates/project/project.py]: Fixed.

**Pull request closed without merging**